### ISSUE TITLE
The docs won't display `Available values` while using function choices in django-filter

### ISSUE THREAD

**Issue by gotounix** [tfranzel/drf-spectacular#1260]
**Describe the bug**

While using `choice_callable` to set `choices` for `ChoiceFilter`, the `Available values` won't display.

Only using `filters.ChoiceFilter(choices=[('A', 'a'), ('b', 'b')])`, the `Available values` will display.

![image](https://github.com/tfranzel/drf-spectacular/assets/18606632/a74df50b-21a5-4bf7-9d48-6fdc63c6aa21)

**To Reproduce**

```
def choice_callable():
    from django_celery_beat.admin import TaskSelectWidget
    return [('A', 'a'), ('b', 'b')]

class askFilter(filters.FilterSet):
    task = filters.ChoiceFilter(choices=choice_callable)
    #  task = filters.ChoiceFilter(choices=[('A', 'a'), ('b', 'b')])
```

**Expected behavior**

Display `Available values` while using `choice_callable`.

**Comment by tfranzel** [comment]
Hi, this is intended behavior.

https://github.com/tfranzel/drf-spectacular/blob/cc916372a0e3fafc3af48f94dd1a985f5fc466e9/drf_spectacular/contrib/django_filters.py#L227-L229

This is for 2 reasons:
* It should be possible to generate the schema without a database connection.  (CI/CD pipelines, tests, etc.)
* Because there might be arbitrarily complex&slow queries hidden behind that call that would be executed every time the schema is generated.

**Comment by gotounix** [comment]
@tfranzel 
Is it possible to add a setting to control the enabling or disabling of using a function to generate options in the documentation?

**Comment by tfranzel** [comment]
No. As I said this breaks a core assumption. If we would add a setting for stuff like this we would have 300 setting flags.

However, you are free to subclass the django-filters extension and change the behavior. Just remeber to set the priority of the extension class so that it supercedes the original extension.
